src/main/java/org/elasticsearch/common/geo/GeoPoint.java

```diff
@@ -119,15 +130,15 @@ public boolean equals(Object o) {
     public int hashCode() {
         int result;
         long temp;
-        temp = lat != +0.0d ? Double.doubleToLongBits(lat) : 0L;
+        temp = y != +0.0d ? Double.doubleToLongBits(y) : 0L;
         result = (int) (temp ^ (temp >>> 32));
-        temp = lon != +0.0d ? Double.doubleToLongBits(lon) : 0L;
+        temp = x != +0.0d ? Double.doubleToLongBits(x) : 0L;
         result = 31 * result + (int) (temp ^ (temp >>> 32));
         return result;
     }
 
     public String toString() {
-        return "[" + lat + ", " + lon + "]";
+        return "[" + y + ", " + x + "]";
```

Comment:
To comply with GeoJSON, absolutely.  I was a little concerned about changing order for sake of the existing Java API. If ES users are doing anything upstream with that string and we reverse order on them... But I do agree that this should comply with the GeoJSON spec, and so should they.

For consistency we should probably revisit the docs at http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/mapping-geo-point-type.html and the ```geo_point``` type in general.  The order was originally set to comply with "natural language"  ```lat/lon``` ordering to make life easy.